compaction/compaction.cc

```diff
@@ -229,10 +229,10 @@ static std::vector<shared_sstable> get_uncompacting_sstables(const table_state&
     auto all_sstables = boost::copy_range<std::vector<shared_sstable>>(*table_s.main_sstable_set().all());
     auto& compacted_undeleted = table_s.compacted_undeleted_sstables();
     all_sstables.insert(all_sstables.end(), compacted_undeleted.begin(), compacted_undeleted.end());
-    boost::sort(all_sstables, [] (const shared_sstable& x, const shared_sstable& y) {
+    std::ranges::sort(all_sstables, [] (const shared_sstable& x, const shared_sstable& y) {
         return x->generation() < y->generation();
     });
```

Comment:
You could use the projection parameter: `std::ranges::sort(all_sstables, std::less<>(), std::mem_fn(&sstable::generation));`

Reply:
done in v2